### PR TITLE
Feature : Add cancellable IslandBanEvent

### PR DESCRIPTION
**Pull request by NewZAZ**
Add posibility to cancel IslandBanEvent !

### CODE REVIEW

File: src/main/java/com/bgsoftware/superiorskyblock/island/IslandUtils.java [hunk 1]
```diff
@@ -257,7 +257,9 @@ public static boolean checkBanRestrictions(SuperiorPlayer superiorPlayer, Island
     }
 
     public static void handleBanPlayer(SuperiorPlayer caller, Island island, SuperiorPlayer target) {
-        plugin.getEventsBus().callIslandBanEvent(caller, target, island);
+        boolean isCancelled = plugin.getEventsBus().callIslandBanEvent(caller, target, island);
```
Comment (OmerBenGera) [inline src/main/java/com/bgsoftware/superiorskyblock/island/IslandUtils.java]: Can be optimized into one if line:
```
if(call ban event)
    return;
```

**Review by OmerBenGera — CHANGES_REQUESTED** [review]

File: src/main/java/com/bgsoftware/superiorskyblock/island/IslandUtils.java (hunk 1 above)
Comment (NewZAZ) [reply src/main/java/com/bgsoftware/superiorskyblock/island/IslandUtils.java]: good :)

**Review by OmerBenGera — APPROVED** [review]

File: src/main/java/com/bgsoftware/superiorskyblock/core/events/EventsBus.java [hunk 2]
```diff
@@ -67,10 +67,11 @@ public boolean callBlockUnstackEvent(Block block, Player player, int originalAmo
                 "blockunstackevent");
     }
 
-    public void callIslandBanEvent(SuperiorPlayer superiorPlayer, SuperiorPlayer targetPlayer, Island island) {
+    public boolean callIslandBanEvent(SuperiorPlayer superiorPlayer, SuperiorPlayer targetPlayer, Island island) {
         if (!plugin.getSettings().getDisabledEvents().contains("islandbanevent")) {
```
Comment (OmerBenGera) [inline src/main/java/com/bgsoftware/superiorskyblock/core/events/EventsBus.java]: Should be changed to the format of calling a cancellable event. For example:
```java
    public boolean callAttemptPlayerSendMessageEvent(SuperiorPlayer receiver, String messageType, Object... args) {
        return callEvent(() -> new AttemptPlayerSendMessageEvent(receiver, messageType, args), "attemptplayersendmessageevent");
    }
```

**Review by OmerBenGera — CHANGES_REQUESTED** [review]

**Comment by NewZAZ** [comment]
Good now ?

**Review by OmerBenGera — APPROVED** [review]

**Pull request merged**